api/src/test/java/org/openmrs/module/appframework/service/AppFrameworkServiceImplTest.java

```diff
@@ -53,23 +53,26 @@
 
     private Extension ext5;
 
-
     @Before
     public void setUp() throws Exception {
 
+    	// clear up from previous tests
```

Comment:
I'm not sure if it's necessary to make a change in core, the only reason we has @DirtiesContext was to have the apps cleaned up as the application context gets reloaded, since you got rid of @DirtiesContext annotation, a manual clean up of the apps seems like the obvious solution which you are doing, all we are saying is that it should happen after each test and not before and my guess is that this has nothing to do with authentication unless may be there is something else you're doing.